SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -527,6 +527,12 @@ void USpatialSender::SendComponentInterestForActor(USpatialActorChannel* Channel
 {
 	checkf(!NetDriver->IsServer(), TEXT("Tried to set ComponentInterest on a server-worker. This should never happen!"));
 
+	if (!bNetOwned)
+	{
+		// Ensure we don't retain stale component data when losing ownership
+		NetDriver->Receiver->ClearPendingOwnerOnlyComponentsForEntity(EntityId);
```

Comment:
Did you find this was actually clearing components when running without result types?

Reply:
I haven't observed it, tbh. I was following @mironec 's suggestion to solve it this way. Not entirely sure how a test case would have to look like to make sure this path is taken.

Reply:
I think this should fire in a normal multi-client game because owner_only components only stop being sent after the initial entity is checked out.

Reply:
Hm, played a 3 player example project on a local deployment and this didn't fire

Reply:
Have you turned off result types? Turning them off is kind of annoying, you will have to modify the SpatialGDKSettings.ini file. I think this needs to be tested before it goes in.